### ISSUE TITLE
kms.list_resource_tags does not support KeyArn parameter

### ISSUE THREAD

**Issue by rtoma** [spulec/moto#4312]
Hi,

Expected behaviour: call `kms.list_resource_tags(keyARN)` and get the tags back (if the key was created before).

Actual behaviour:
```
botocore.errorfactory.NotFoundException: An error occurred (NotFoundException) when calling
  the ListResourceTags operation: The request was rejected because the specified entity or resource
  could not be found.
```

Boto3 docs https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/kms.html#KMS.Client.list_resource_tags says you can pass KeyID or KeyARN as parameter.

I traced the `moto.kms.models.list_resource_tags` method and I see it does a dict.get on a dict keyed by KeyID. So, it makes sense a lookup by KeyARN fails.

Thanks again for developing moto!

**Comment by bblommers** [comment]
Hi @rtoma,thanks for raising this! Marking it as a bug, as we should support both options.

**Comment by bblommers** [comment]
This should be fixed as of moto >= 2.2.8.dev23 - let us know whether we've missed anything.
